### ISSUE TITLE
Is SetOrderedFactoryAsCurrentAndGetResetter safe for parallel tests?

### ISSUE THREAD

**Issue by TimLovellSmith** [FakeItEasy/FakeItEasy#498]
(In the scenario of running test cases in parallel, that is.)

To me it looks not so, due to dependency on static field currentAsserterFactoryField but I might be missing some constraint on how it can be used in practice.

**Comment by blairconrad** [comment]
@TimLovellSmith, I'm inclined to agree. From a quick eyeballing, there's a good chance it wouldn't work.
To tell you the truth, I've a sneaking suspicion that nothing `Scope`y will work.
Have you encountered any problems?

**Comment by TimLovellSmith** [comment]
No, I just noticed this while reviewing FakeItEasy code while having another look at issue #484 - which is still stumping me.

**Comment by TimLovellSmith** [comment]
Aha, I might have figured out 484 now. Anyway this one is totally unimportant to me for now. Voting to postpone/close :dancer:

**Comment by blairconrad** [comment]
Well, technically it was a question, and you seem content with what we've come up with, so I'm on board. Closing.
